### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by IgorBochkor**
added solution

### CODE REVIEW

File: src/main/java/com/dev/cinema/annotation/PasswordValidation.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    Class<?>[] groups() default {};
+
+    Class<? extends Payload>[] payload() default {};
+
+    @Target({ElementType.TYPE})
+    @Retention(RetentionPolicy.RUNTIME)
+    @interface List {
+        PasswordValidation[] value();
+    }
```
Comment (yurwing) [inline src/main/java/com/dev/cinema/annotation/PasswordValidation.java]: you can delete this

**Review by yurwing — APPROVED** [review]

File: src/main/java/com/dev/cinema/config/MyWebAppInitializer.java [hunk 2]
```diff
@@ -5,7 +5,8 @@
 public class MyWebAppInitializer extends AbstractAnnotationConfigDispatcherServletInitializer {
     @Override
     protected Class<?>[] getRootConfigClasses() {
-        return new Class[]{com.dev.cinema.config.AppConfig.class};
+        return new Class[]{com.dev.cinema.config.AppConfig.class,
+                com.dev.cinema.config.SecurityConfig.class};
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/config/MyWebAppInitializer.java]:
```suggestion
        return new Class[]{AppConfig.class, SecurityConfig.class};
```
can we make imports?

File: src/main/java/com/dev/cinema/controllers/OrderController.java [hunk 3]
```diff
@@ -30,13 +32,17 @@ public OrderController(OrderService orderService, ShoppingCartService shoppingCa
...
+        orderService.completeOrder(shoppingCartService.getByUser(user));
     }
 
     @GetMapping
-    public List<OrderResponseDto> get(@RequestParam Long userId) {
-        return orderService.getOrdersHistory(userService.getById(userId)).stream()
+    public List<OrderResponseDto> get(Authentication auth) {
+        UserDetails userDetails = (UserDetails) auth.getPrincipal();
+        User user = userService.findByEmail(userDetails.getUsername()).get();
+        return orderService.getOrdersHistory(userService.getById(user.getId())).stream()
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/OrderController.java]:
```suggestion
        return orderService.getOrdersHistory(user).stream()
```

File: src/main/java/com/dev/cinema/controllers/ShoppingCartController.java [hunk 4]
```diff
@@ -31,14 +34,18 @@ public ShoppingCartController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/movie-sessions")
-    public void create(@RequestParam Long userId, @RequestParam Long movieSessionId) {
+    public void create(Authentication auth, @RequestParam Long movieSessionId) {
+        UserDetails userDetails = (UserDetails) auth.getPrincipal();
+        User user = userService.findByEmail(userDetails.getUsername()).get();
         shoppingCartService.addSession(movieSessionService.getById(movieSessionId),
-                userService.getById(userId));
+                userService.getById(user.getId()));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/ShoppingCartController.java]: same

File: src/main/java/com/dev/cinema/controllers/ShoppingCartController.java [hunk 5]
```diff
@@ -31,14 +34,18 @@ public ShoppingCartController(ShoppingCartService shoppingCartService,
...
     }
 
     @GetMapping("/by-user")
-    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
-        ShoppingCart byUser = shoppingCartService.getByUser(userService.getById(userId));
+    public ShoppingCartResponseDto getByUser(Authentication auth) {
+        UserDetails userDetails = (UserDetails) auth.getPrincipal();
+        User user = userService.findByEmail(userDetails.getUsername()).get();
+        ShoppingCart byUser = shoppingCartService.getByUser(userService.getById(user.getId()));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/ShoppingCartController.java]: same

File: src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java [hunk 6]
```diff
@@ -1,8 +1,14 @@
 package com.dev.cinema.model.dto;
 
+import javax.validation.constraints.Min;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java]: wrappers are not supported by this annotation

File: src/main/java/com/dev/cinema/annotation/PasswordValidation.java (hunk 1 above)
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/annotation/PasswordValidation.java]: I agree that we may remove this

File: src/main/java/com/dev/cinema/util/PasswordValidator.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
+package com.dev.cinema.util;
+
+import com.dev.cinema.annotation.PasswordValidation;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class PasswordValidator
+        implements ConstraintValidator<PasswordValidation, Object> {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/util/PasswordValidator.java]: redundant empty line

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/annotation/PasswordValidation.java (hunk 1 above)
Comment (IgorBochkor) [reply src/main/java/com/dev/cinema/annotation/PasswordValidation.java]: agree

File: src/main/java/com/dev/cinema/config/MyWebAppInitializer.java (hunk 2 above)
Comment (IgorBochkor) [reply src/main/java/com/dev/cinema/config/MyWebAppInitializer.java]: fixed

File: src/main/java/com/dev/cinema/controllers/OrderController.java (hunk 3 above)
Comment (IgorBochkor) [reply src/main/java/com/dev/cinema/controllers/OrderController.java]: fixed

File: src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java (hunk 6 above)
Comment (IgorBochkor) [reply src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java]: fixed

File: src/main/java/com/dev/cinema/annotation/PasswordValidation.java (hunk 1 above)
Comment (IgorBochkor) [reply src/main/java/com/dev/cinema/annotation/PasswordValidation.java]: fixed

**Comment by IgorBochkor** [comment]
"равлика" також не треба ставити в назву коміта, не тільки в коментарях, буду знати)

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**